sentry.client.config.ts

```diff
@@ -1,17 +1,68 @@
-import { SENTRY_DSN, SENTRY_ENABLE } from '@/next.constants.mjs';
+import {
+  Dedupe,
+  Breadcrumbs,
+  HttpContext,
+  LinkedErrors,
+  BrowserClient,
+  getCurrentHub,
+  defaultStackParser,
+  makeFetchTransport,
+} from '@sentry/nextjs';
 
-// This lazy-loads Sentry on the Browser
-import('@sentry/nextjs').then(({ init }) =>
-  init({
-    // Only run Sentry on Vercel Environment
-    enabled: SENTRY_ENABLE,
-    // Tell Sentry where to send events
-    dsn: SENTRY_DSN,
-    // Disable Sentry Tracing as we don't need to have it
-    // as Vercel already does Web Vitals and Performance Measurement on Client-Side
-    enableTracing: false,
-    // We only want to capture errors from _next folder on production
-    // We don't want to capture errors from preview branches here
-    allowUrls: ['https://nodejs.org/_next'],
-  })
-);
+import {
+  SENTRY_DSN,
+  SENTRY_ENABLE,
+  SENTRY_CAPTURE_RATE,
+  SENTRY_TUNNEL,
+} from '@/next.constants.mjs';
+
+// This creates a custom Sentry Client with minimal integrations
+export const sentryClient = new BrowserClient({
+  // Only run Sentry on Vercel Environment
+  enabled: SENTRY_ENABLE,
+  // Provide Sentry's Secret Key
+  dsn: SENTRY_DSN,
+  // Sentry's Error Transport Mechanism
+  transport: makeFetchTransport,
+  // Sentry's Stack Trace Parser
+  stackParser: defaultStackParser,
+  // All supported Integrations by us
+  integrations: [
+    new Dedupe(),
+    new HttpContext(),
+    new Breadcrumbs(),
+    new LinkedErrors(),
+  ],
+  // We only want to capture errors from _next folder on production
+  // We don't want to capture errors from preview branches here
+  allowUrls: ['https://nodejs.org/', /^https:\/\/.+\.vercel\.app/],
```

Comment:
if this is an allow list, do you need to negate the vercel app urls at all?

Reply:
sorry what? Didn't get the comment

Reply:
oh i misread the regex - its not a negation - it's a starting character.... but i guess the question remains, what urls are we expecting to serve from vercel.app ?

Reply:
Our preview branches...?

Reply:
Guess I'm confused. The comment for the line says we `dont` want them.

Reply:
Let me fix that comment